z/buffer.go

```diff
@@ -183,56 +143,72 @@ func (b *Buffer) Bytes() []byte {
 	return b.buf[b.padding:off]
 }
 
-func (b *Buffer) AutoMmapAfter(size int) {
-	b.autoMmapAfter = size
-}
-
 // Grow would grow the buffer to have at least n more bytes. In case the buffer is at capacity, it
 // would reallocate twice the size of current capacity + n, to ensure n bytes can be written to the
 // buffer without further allocation. In UseMmap mode, this might result in underlying file
 // expansion.
 func (b *Buffer) Grow(n int) {
-	// In this case, len and cap are the same.
 	if b.buf == nil {
 		panic("z.Buffer needs to be initialized before using")
 	}
-	if b.maxSz-int(b.offset) < n {
-		panic(fmt.Sprintf("Buffer max size exceeded: %d."+
-			" Offset: %d. Grow: %d", b.maxSz, b.offset, n))
+	if b.maxSz != 0 && int(b.offset)+n > b.maxSz {
```

Comment:
When can b.maxSz be 0? It should always have a valid value.

Reply:
I've kept the zero value as an "unbounded" buffer. The earlier API set `maxSz = math.MaxInt32` (roughly 2.1GB) by default. Should I change it back?